Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Symfony;
+
+use SplFileInfo;
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Make sure there is one blank line above and below a method.
+ */
+final class MethodSeparationFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(SplFileInfo $file, Tokens $tokens)
+    {
+        $tokenSize = $tokens->getSize();
+        for ($index = 0; $index < $tokenSize; ++$index) {
+            if (false === $tokens[$index]->isClassy()) {
+                continue;
+            }
+
+            // figure out where the classy opens
+            $classOpen = $tokens->getNextTokenOfKind($index, array('{'));
+            if (null === $classOpen) {
+                throw new \RuntimeException('Classy opening not found.');
+            }
+
+            // figure out where the classy ends
+            $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classOpen, true);
+            if (null === $classEnd) {
+                throw new \RuntimeException('Classy closing not found.');
+            }
+
+            if ($tokens[$index]->isGivenKind(T_CLASS)) {
+                $index = $this->fixClass($tokens, $classOpen, $classEnd);
```

Comment:
I think it could be simpler if we decide to iterate main loop backward.
Then we could just go through index and have no need for jumps like that and recalculating `$classEnd` in helpers